### PR TITLE
chore: use UV mapping

### PR DESCRIPTION
**Pull request by jo-lyn**
Use UV mapping with texture atlas so that we can programmatically set highlighted texture on hover.

### Current system for texture atlas
- Each texture atlas contains a normal texture and highlighted texture
- Currently one texture maps to all 6 faces of a block. The mapping coordinates will change when we introduce different textures for each faces.
- UV mapping is only in effect on start. We can still drag materials to prefabs in editor but it will be overridden by the material variable in the UVMap script when we start the game.

For now only grass prefab is updated with the texture atlas.

Reference: https://www.youtube.com/watch?v=EWQpo4sjuxw

### CODE REVIEW

**Comment by nicholaschuayunzhi** [comment]
Currently there's no hover logic. I will implement it asap.

Are you able to expose a method that lets me change a specific face to a highlighted texture?

File: Assets/Scripts/UVMap.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        blockUVs[17] = new Vector2(umax, vmax);
+        blockUVs[18] = new Vector2(umin, vmax);
+        blockUVs[19] = new Vector2(umax, vmin);
+        blockUVs[20] = new Vector2(umin, vmin);
+        blockUVs[21] = new Vector2(umax, vmax);
+        blockUVs[22] = new Vector2(umin, vmax);
+        blockUVs[23] = new Vector2(umax, vmin);
+
+        this.GetComponent<Renderer>().material = material;
+        this.GetComponent<MeshFilter>().mesh.uv = blockUVs;
```
Comment (Rinder5) [inline Assets/Scripts/UVMap.cs]: GetComponent is apparently a pretty computationally intensive method. Possible to grab it only once and store a reference somewhere?

**Comment by nicholaschuayunzhi** [comment]
@jo-lyn  i've added my changes. Currently in the lighting scene, the blocks use the grass atlas texture and this what happens when playing: 
image

Have to update the other texture atlases

**Comment by nicholaschuayunzhi** [comment]
@jo-lyn Ok that one can be done in a separate PR. I have removed the UVMaps in the lightign scene for stone and sand blocks so it looks fine.

**Review by Rinder5 — APPROVED** [review]
Looks good! Thanks for your hard work.

**Pull request merged**